media-libs/libheif/libheif-1.10.0.ebuild

```diff
@@ -0,0 +1,67 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools xdg multilib-minimal
+
+if [[ ${PV} == "9999" ]] ; then
+	EGIT_REPO_URI="https://github.com/strukturag/libheif.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/strukturag/libheif/releases/download/v${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+fi
+
+DESCRIPTION="ISO/IEC 23008-12:2017 HEIF file format decoder and encoder"
+HOMEPAGE="https://github.com/strukturag/libheif"
+
+LICENSE="GPL-3"
+SLOT="0/1.10"
+IUSE="gdk-pixbuf go test +threads"
+REQUIRED_USE="test? ( go )"
+
+RESTRICT="!test? ( test )"
+
+BDEPEND="test? ( dev-lang/go )"
+DEPEND="
+	media-libs/dav1d:=[${MULTILIB_USEDEP}]
+	>=media-libs/libaom-2.0.0:=[${MULTILIB_USEDEP}]
+	media-libs/libde265:=[${MULTILIB_USEDEP}]
+	media-libs/libpng:0=[${MULTILIB_USEDEP}]
+	media-libs/x265:=[${MULTILIB_USEDEP}]
+	media-video/rav1e:=
```

Comment:
I am not sure about this, see https://github.com/strukturag/libheif/blob/v1.10.0/CMakeLists.txt#L132 -- libheif will only need an AVIF encoder. But it can be libaom or rav1e.

Also dav1d and libaom are interchangeable.

libde265 is also optional like x265.

Reply:
Yes, you're right. The thing is, build system doesn't have configure switches to choose between all encoders/decoders. I've added some USE flags to chose between different libraries, and adjusted the mask for `rav1e` support